cpu/sam21_common/Makefile.include

```diff
@@ -1,6 +1,8 @@
-# export the CPU family so we can differentiate between them in the code
-FAM = $(shell echo $(CPU_FAM) | tr 'a-z-' 'A-Z_')
-export CFLAGS += -DCPU_FAM_$(FAM)
+# Define the CPU family so we can differentiate between them in the code
+CFLAGS += -DCPU_FAM_$(shell echo $(CPU_FAM) | tr 'a-z-' 'A-Z_')
+
+# Define the SAM0 family so we can differentiate between them in the code
+CFLAGS += -DSAM0_FAM_$(shell echo $(SAM0_FAM) | tr 'a-z-' 'A-Z_')
```

Comment:
> I looked at it, but the problem is that `CPU_FAM` is currently set in `cpu/samd21/Makefile.include`. And that one is shared for samd21 and samr21 boards. We need a parameter to distinguish between these two.

Why? We don't have to use the dispatching from CMSIS.

We currently support one samd21 CPU (SAMR21G18A), and that number might
rise to 5-10 in the mid-term.

IMHO we should just select the right cmsis header directly (in cpu_conf.h).

Reduces "-DCPU_FAM=samd21 -DSAM0_FAM=samr21 -D__SAMR21G18A__
-DCPU_FAM_SAMD21 -DSAM0_FAM_SAMR21" by 40%. ;)